packages/icons/src/Icon.js

```diff
@@ -1,97 +1,52 @@
 import React from 'react'
 import upperFirst from 'lodash.upperfirst'
 import PropTypes from 'prop-types'
-
 import * as icons from './index'
 
 const Icon = ({ name, title, desc, titleId, descId, ...props }) => {
   const Component = icons[name] || icons[upperFirst(name)]
-  if (!Component) return false
-
-  /**
-   *  To support older browsers, make sure 'titleId' is passed along with 'title' props and
-   * 'descId' with 'desc' props
-   * Also, <desc> element should be followed by a <title> tag for <svg> elements
-   * */
-  if (!!title) {
-    let ariaLabelledBy = ''
-    props['aria-hidden'] = 'false'
-
-    if (!!titleId) {
-      ariaLabelledBy = titleId
-    }
 
-    if (!!desc && !!descId) {
-      ariaLabelledBy = `${ariaLabelledBy} ${descId}`
-    }
-
-    if (!!ariaLabelledBy) {
-      props['aria-labelledby'] = ariaLabelledBy
-    }
-  }
+  const description = desc && descId ? descId : ''
+  const ariaLabelledBy =
+    title && titleId ? `${titleId} ${description}` : description
 
   return (
-    <Component
-      title={title}
-      desc={desc}
-      titleId={titleId}
-      descId={descId}
-      {...props}
-    />
+    Component && (
+      <Component
+        title={title}
+        desc={desc}
+        titleId={titleId}
+        descId={descId}
+        aria-hidden={!ariaLabelledBy}
+        aria-labelledby={ariaLabelledBy}
+        {...props}
+      />
+    )
   )
 }
 
+Icon.isIcon = true
+Icon.displayName = 'Icon'
+
 Icon.defaultProps = {
   size: 24,
-  'aria-hidden': 'true',
-  focusable: 'false',
-  tabIndex: '-1'
+  tabIndex: -1,
+  focusable: false,
+  'aria-hidden': true
 }
 
-Icon.displayName = 'Icon'
-Icon.isIcon = true
-
 Icon.propTypes = {
   name: (props, key, componentName) => {
     const name = props[key]
     if (!icons[name] && !icons[upperFirst(name)]) {
       return new Error(
         `Unknown name prop \`${name}\` supplied to \`${componentName}\``
       )
-    } else if (!icons[name]) {
-      return new Error(
-        `Icon name prop should be uppercase.\n` +
-          `Use \`${upperFirst(name)}\` instead of ${name}.`
```

Comment:
I ran into a case where this warning actually caused a production bug. For some reason locally using uppercase names was fine but we found that production only accepted lowercase icon names.